benchmark/src/worker.rs

```diff
@@ -72,17 +79,26 @@ impl WorkerService for Worker {
                 None => return sink.close().await,
                 Some(arg) => arg,
             };
+            #[cfg(feature = "protobuf-codec")]
             let cfg = arg.get_setup();
+            #[cfg(feature = "protobufv3-codec")]
+            let cfg = arg.setup();
             info!("receive client setup: {:?}", cfg);
             let mut client = Client::new(cfg);
             sink.send((ClientStatus::default(), WriteFlags::default()))
                 .await?;
             while let Some(arg) = stream.try_next().await? {
+                #[cfg(feature = "protobuf-codec")]
                 let mark = arg.get_mark();
+                #[cfg(feature = "protobufv3-codec")]
```

Comment:
How about using a trait to unify them?

```
trait Accessor {
    type Item;
    fn get(&self) -> &Self::Item;
    fn r#mut(&mut self) -> &mut Self::Item;
}

impl<T> Accessor for Option<T> {
    type Item = T;
    xxx
}

impl<T> Accessor for SinglePtrField<T> {
    type Item = T;
    xxx
}
```
?

Reply:
Hi, I'm confused on how to do this. My understanding is that you want a generic trait to implement both the types used in v2 and v3, so they can be accessed with a one-liner `.get()` or `.mut()` regardless of the version.
However in this case, the methods have different names `.get_mark()` vs `.mark()` so we'd still need the distinction ? What am I missing ? thanks

Reply:
So they should be able to be accessed by `.mark.get()`. The accessors `get_xxx` and `xxx` are used to handle the case that `xxx` is missing. It can be implemented in the `trait Accessor` directly.

Reply:
I see, let's keep it the current way for now.

Reply:
Sounds good. Then please let me know if you have other comments, I believe all your previous comments are addressed.